pkg/services/authn/grpcutils/config.go

```diff
@@ -0,0 +1,35 @@
+package grpcutils
+
+import (
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type GrpcClientConfig struct {
+	Token            string
+	TokenExchangeURL string
+	TokenNamespace   string
+}
+
+func ReadGrpcClientConfig(cfg *setting.Cfg) *GrpcClientConfig {
+	section := cfg.SectionWithEnvOverrides("grpc_client_authentication")
+
+	return &GrpcClientConfig{
+		Token:            section.Key("token").MustString(""),
+		TokenExchangeURL: section.Key("token_exchange_url").MustString(""),
+		TokenNamespace:   section.Key("token_namespace").MustString("stack-" + cfg.StackID),
+	}
+}
+
+type GrpcServerConfig struct {
+	SigningKeysURL   string
+	AllowedAudiences []string
+}
+
+func ReadGprcServerConfig(cfg *setting.Cfg) (*GrpcServerConfig, error) {
```

Comment:
No error needed anymore.
```suggestion
func ReadGprcServerConfig(cfg *setting.Cfg) (*GrpcServerConfig) {
```

Reply:
error is needed, but function name has a typo:
```suggestion
func ReadGrpcServerConfig(cfg *setting.Cfg) (*GrpcServerConfig, error) {
```

Reply:
You are write, seems like we reintroduces modes :+1: